src/script/conversation/ConversationState.ts

```diff
@@ -111,14 +103,28 @@ export class ConversationState {
           ConnectionStatus.PENDING,
         ];
 
+        const isCleared = conversationEntity.is_cleared();
+        const isRemoved = conversationEntity.removed_from_conversation();
+
         if (
           isSelfConversation(conversationEntity) ||
           states_to_filter.includes(conversationEntity.connection().status())
         ) {
           return false;
         }
 
-        return !(conversationEntity.is_cleared() && conversationEntity.removed_from_conversation());
+        if (isMLSConversation(conversationEntity)) {
+          const isMLSConversationEstablished = useMLSConversationState
+            .getState()
+            .isEstablished(conversationEntity.groupId);
+
+          //conversation doesn't need to be established if user was removed
+          if (!isRemoved && !isMLSConversationEstablished) {
+            return false;
+          }
```

Comment:
This is the main change, the rest is just refactoring of where things are computed.